### PR TITLE
Added variables for reward trackings

### PR DESCRIPTION
**Pull request by baijiuyang**
Added a dictionary to save train reward, evaluation reward, and losses.
Added code to track train reward and evaluation reward, but I don't know how to track the losses of critic, value, and actor using the History objects.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -90,7 +90,15 @@ def main():
     gac = GACAgent(args)
 
     state = env.reset()
-
+    
+    results_dict = {'train_rewards': []
+                    'eval_rewards': [],
+                    'actor_losses': [],
+                    'value_losses': [],
+                    'policy_losses': [],
```
Comment (axlbrown) [inline main.py]: I think policy_losses really means critic_losses?

**Review by axlbrown — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (baijiuyang) [reply main.py]: Yes, it was typo. I'll fix it.

File: main.py [hunk 2]
```diff
@@ -101,11 +109,27 @@ def main():
         action = gac.get_action(tf.convert_to_tensor([state]))
         next_state, reward, is_terminal, _ = env.step(action)
         gac.store_transitions(state, action, reward, next_state, is_terminal)
-        state = env.reset() if is_terminal else next_state
-
+        
+        # check if game is terminated to decide how to update state, episode_steps, episode_rewards
+        if is_terminal:
+            state = env.reset()
+            results_dict['train_rewards'].apped((t, episode_rewards / episode_steps))
```
Comment (axlbrown) [inline main.py]: typo: append

File: main.py [hunk 3]
```diff
@@ -101,11 +109,27 @@ def main():
...
         if gac.replay.size >= args.batch_size:
-            gac.train_one_step()
-
+            critic_history, value_history, actor_history = gac.train_one_step()
+            # I don't know how to use History objects....
+            # Somebody who knows please append the losses of critc, value and actor to results_dict
+            
+        # evaluate
+        if t % args.eval_freq == 0:
+            results_dict['eval_rewards'].apped((t, evaluate_policy(gac, env, args.eval_episodes)))
```
Comment (axlbrown) [inline main.py]: typo: append

**Review by axlbrown — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (baijiuyang) [reply main.py]: fixed

File: main.py (hunk 3 above)
Comment (baijiuyang) [reply main.py]: Good catch! Fixed.

**Review by axlbrown — COMMENTED** [review]
lgtm

**Review by axlbrown — APPROVED** [review]

**Pull request merged**